src/component/MovieForm/MovieForm.js

```diff
@@ -1,60 +1,140 @@
+              </label>
+              <label key="poster_path" className="movie-form_input-label">
+                {MOVIE_URL}
+                <input
+                  type="text"
+                  name="poster_path"
+                  className="movie-form_input"
+                  {...register("poster_path", { 
+                    required: 'Field is required',
+                    pattern: {
+                      value: /^(ftp|http|https):\/\/[^ "]+$/,
+                      message: 'Please enter a valid URL',
+                    }, 
+                  
+                  })}
+                />
+                  {errors.poster_path && <span className="error-message">{errors.poster_path.message}</span>}
+              </label>
+              <label key="vote_average" className="movie-form_input-label">
+                {RATING}
+                <input
+                  type="number"
+                  name="vote_average"
+                  className="movie-form_input"
+                  {...register("vote_average", { required: 'Field is required' })}
+                />
+                  {errors.vote_average && <span className="error-message">{errors.vote_average.message}</span>}
+              </label>
+              <label key="genres" className="movie-form_input-label">
+                {GENRE}
+                <input
+                  type="text"
+                  name="genres"
+                  className="movie-form_input"
+                  {...register("genres", { required: 'Field is required' })}
+                  onChange={handleGenreInputChange}
+                />
+                  {errors.genres && <span className="error-message">{errors.genres.message}</span>}
+              </label>
+              <label key="runtime" className="movie-form_input-label">
+                {DURATION}
                 <input
                   type="text"
-                  name={name}
+                  name="runtime"
                   className="movie-form_input"
-                  value={formData[name]}
-                  onChange={handleInputChange}
+                  {...register("runtime", {
+                     required: 'Field is required', 
+                     pattern: {
+                      value: /^(\d+\.?\d*|\.\d+)$/,
+                      message: 'Please enter a valid duration',
+                    },
+                    })}
                 />
+                  {errors.runtime && <span className="error-message">{errors.runtime.message}</span>}
               </label>
-            ))}
           </div>
           <label className="movie-form_input-label movie-form_textarea">
             {OVERVIEW}
             <textarea
               className="wider-input movie-form_input"
               type="text"
               name="overview"
-              value={formData.overview}
-              onChange={handleInputChange}
+              {...register('overview', { required: 'Field is required' })}
```

Comment:
Avoid magic strings "Field is Required" , move to constants file.

Reply:
Fixed as part of a5b5e33